R-package/R/callback.R

```diff
@@ -97,13 +99,15 @@ cb.reset.parameters <- function(new_params) {
     if (!is.null(env$model)) {
       env$model$reset_parameter(params = pars)
     }
+    
+    return(invisible(NULL))
```

Comment:
I think this place MIGHT be related to the test failures you saw. `env$model$reset_parameter(params = pars)` returns a `Booster`, so the current behavior of this function is that that booster is returned if this code runs:

```r
    if (!is.null(env$model)) {
      env$model$reset_parameter(params = pars)
    }
```

This `return(invisible(NULL))` is correct, but please also add a `return()` inside that if statement above.

This kind of thing is why this PR is so valuable! To find places like this where `{lightgbm}` functions' return values are non-obvious. So thank you!

Reply:
It turns out that the test failures were caused by a misplaced `return()` statement resulted from the merge. I have fixed it in commit 1f8661a .